lightning/src/events/bump_transaction.rs

```diff
@@ -222,6 +247,17 @@ pub enum BumpTransactionEvent {
 	/// [`EcdsaChannelSigner::sign_holder_htlc_transaction`]: crate::sign::EcdsaChannelSigner::sign_holder_htlc_transaction
 	/// [`HTLCDescriptor::tx_input_witness`]: HTLCDescriptor::tx_input_witness
 	HTLCResolution {
+		/// The unique identifier for the claim of the HTLCs in the confirmed commitment
+		/// transaction.
+		///
+		/// Any future instances of `HTLCResolution` bump events that share the same `claim_id` must
```

Comment:
So iirc we don’t implement splitting our package based on timelocks, only at detection when it’s near expiration (`CLTV_SHARED_CLAIM_BUFFER`). We still aggregate preimage claims so if an offered HTLC is individually claimed by a counterparty timeout transaction, there is no more risk of UTXOs double-spending, so we can free the “UTXO lock” and re-assign the UTXOs to the new broadcast of the set of HTLCs to claim ? The `ClaimId` could be based on the exact same set of HTLCs no, or it’s not an API guarantee ?

Reply:
> Having them be dynamic (as originally done when the PR was opened) would undo the work that resulted from the discussion in https://github.com/lightningdevkit/rust-lightning/pull/2089#discussion_r1182018351.

Yeah, there might no need of dynamically computing `ClaimId`, as long as we can give other information to the `CoinSelectionSource` such as the value associated to a set of HTLCs, and decide on this to double-spend UTXOs.

> As we discussed in other threads, we don't want to be too opinionated here so we defer to the user instead. The pending package X will have its own unique claim_id and will be handled once its bump event is dispatched. If the CoinSelectionSource implementation doesn't have any free UTXOs, they can double spend package(s) of their choosing.

Unlocking UTXOs of _static_ `ClaimId` that have been partially double-spent or reorgs out would be still great from fee-bumping reserve management viewpoint. We can see how users are doing with this current API and improve things over time.